docs/docs/examples/login/flutter.md

```diff
@@ -69,345 +66,112 @@ Basically, there are two major points in this specification:
-            Text(
-              'You are not authenticated.',
-            ),
-          if (!_authenticated && !_busy)
-            ElevatedButton.icon(
-                icon: Icon(Icons.fingerprint),
-                label: Text('login'),
-                onPressed: _authenticate),
-          if (_busy)
-            Stack(
-              children: [
-                Center(child: Text("Busy, logging in.")),
-                Opacity(
-                  opacity: 0.5,
-                  child: Container(
-                    color: Colors.black,
-                  ),
-                ),
-                Center(child: CircularProgressIndicator()),
-              ],
-            ),
-          if (_authenticated && !_busy)
-            Text(
-              'Hello $_username!',
-            ),
-        ],
-      ),
-    ),
-  );
-}
+```dart reference
+https://github.com/zitadel/zitadel_flutter/blob/main/lib/main.dart#L119-L159
 ```
 
 And finally the `_authenticate` method which calls the authorization endpoint,
 then fetches the user info and stores the tokens into the secure storage.
 
-```dart
-Future<void> _authenticate() async {
-  setState(() {
-    _busy = true;
-  });
-
-  try {
-    final result = await _appAuth.authorizeAndExchangeCode(
-      AuthorizationTokenRequest(
-        '<<CLIENT_ID>>', // Client ID of the native application
-        '<<CALLBACK_URL>>', // The registered url from zitadel (e.g. ch.myexample.app://signin)
-        issuer: '<<ISSUER>>', // most of the cases: https:/[your-domain]-[random-string].zitadel.cloud
-        scopes: [
-          'openid',
-          'profile',
-          'email',
-          'offline_access',
-        ],
-      ),
-    );
-
-    final userInfoResponse = await get(
-      Uri.parse('https://[your-instance].zitadel.cloud/oidc/v1/userinfo'),
-      headers: {
-        HttpHeaders.authorizationHeader: 'Bearer ${result.accessToken}',
-        HttpHeaders.acceptHeader: 'application/json; charset=UTF-8'
-      },
-    );
-    final userJson = jsonDecode(utf8.decode(userInfoResponse.bodyBytes));
-
-    await _secureStorage.write(
-        key: 'auth_access_token', value: result?.accessToken);
-    await _secureStorage.write(
-        key: 'refresh_token', value: result?.refreshToken);
-    await _secureStorage.write(key: 'id_token', value: result?.idToken);
-
-    setState(() {
-      _busy = false;
-      _authenticated = true;
-      _username = '${userJson['given_name']} ${userJson['family_name']}';
-    });
-  } catch (e, s) {
-    print('error on authorizeAndExchangeCode token: $e - stack: $s');
-    setState(() {
-      _busy = false;
-      _authenticated = false;
-    });
-  }
-}
+```dart reference
+https://github.com/zitadel/zitadel_flutter/blob/main/lib/main.dart#L45-L117
 ```
 
-Now, you should be able to login with a valid ZITADEL user.
+Note that we have to use our http redirect URL for webapplications or otherwise use our custom scheme for Android and IOS devices.
```

Comment:
```suggestion
Note that we have to use our http redirect URL for web applications or otherwise use our custom scheme for Android and iOS devices.
```